Versionlocks docker & node_exporter packages

alexandre-allard opened the pull request.

NicolasT committed:
ci: fix Buildbot version

Also, install dependencies to build a recent-enough version of Twisted.

alexandre-allard committed:
Change the way node_exporter is versionlocked
Fix versionlock rule
Unlock before package installation and lock it after

Issue: GH-962

alexandre-allard committed:
Add a role to lock docker package version

Issue: GH-962

alexandre-allard committed:
Add a role to unlock docker package version

Issue: GH-962

alexandre-allard committed:
Unlock docker package version before running kubespray

Issue: GH-962

alexandre-allard committed:
Add kubespray-post playbook
This playbook only lock docker package version for the moment

Issue: GH-962

alexandre-allard committed:
Call kubespray-post in deploy & scale

Issue: GH-962

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

NicolasT commented on roles/docker/defaults/main.yml:
```diff
@@ -0,0 +1,7 @@
+
+docker_yum_versionlock_file: /etc/yum/pluginconf.d/versionlock.list
+
+docker_package_names:
```
Should we also lock `container-selinux`? Or is that pinned as reqs of these packages?

alexandre-allard replied:
That's what I did at the beginning, but I've removed it because there are already dependencies in docker-ce & docker-engine RPMs, so no need for that, yum versionlock will prevent from upgrading if dependencies would not be satisfied.

NicolasT commented:
Could you replace the `eve` update commit with 4d097b7c3a5247f821451f0c45ded162e0fee72d? Will ease upmerges.

NicolasT commented:
One more: please add an entry to `ChangeLog.rst` for this change. Other than that, LGTM.

alexandre-allard committed:
Update change logs for issue 962

Issue: GH-962

alexandre-allard commented:
@NicolasT I've cherry-picked 4d097b7c3a5247f821451f0c45ded162e0fee72d and added an entry in `ChangeLog.rst`
I've also changed the way node_exporter is locked, because I found a bug when trying to install the version we want and another version is already installed (there is a comment in the code to explain why it's done like that). Basically, we need to unlock the package before doing the installation, otherwise it would fail if the package is already installed and we want a different version than the one which is installed.

NicolasT reviewed (approved):
@xaf-scality will be happy ;)
